### DISCUSSION TITLE
Is there such a thing as a git diff but for the graphical aspect?

### DISCUSSION THREAD

**Discussion by simkimsia** [terrastruct/d2#1792] [Ideas]
i know i can run git diff or diffmerge on the textual representation via the D2 language.

but i wonder if it's possible to do a diff merge on the graphical representation. Esp since D2 supports bidirectional edits.

**Comment by alixander** [comment]
Oh I like that idea. 

`d2 diff a.d2 b.d2` can show 2 side by side views:
1. `a.d2` but with 
  - everything that was deleted highlighted in red
  - everything that was changed highlighted in yellow
2. `b.d2` but with
  - everything that was added highlighted in green
  - everything that was changed highlighted in yellow

And yes, this would all be possible because of d2oracle (the API that powers bidirectional edits)

**Reply by simkimsia** [reply]
I guess this reply means, it's possible but not live yet. Am I right?

**Reply by alixander** [reply]
Right
